projects/RabbitMQ.Client/client/impl/SocketFrameHandler.cs

```diff
@@ -68,31 +69,16 @@ internal class SocketFrameHandler : IFrameHandler
         private TimeSpan _writeableStateTimeout = TimeSpan.FromSeconds(30);
         private int _writeableStateTimeoutMicroSeconds;
         private readonly ITcpClient _socket;
-        private readonly Stream _reader;
-        private readonly Stream _writer;
-        private readonly ChannelWriter<ReadOnlyMemory<byte>> _channelWriter;
-        private readonly ChannelReader<ReadOnlyMemory<byte>> _channelReader;
-        private readonly Task _writerTask;
+        private readonly PipeReader _pipeReader;
+        private readonly PipeWriter _pipeWriter;
         private readonly object _semaphore = new object();
         private readonly byte[] _frameHeaderBuffer;
         private bool _closed;
 
-        public SocketFrameHandler(AmqpTcpEndpoint endpoint,
-            Func<AddressFamily, ITcpClient> socketFactory,
-            TimeSpan connectionTimeout, TimeSpan readTimeout, TimeSpan writeTimeout)
+        public SocketFrameHandler(AmqpTcpEndpoint endpoint, Func<AddressFamily, ITcpClient> socketFactory, TimeSpan connectionTimeout, TimeSpan readTimeout, TimeSpan writeTimeout)
         {
             Endpoint = endpoint;
-            _frameHeaderBuffer = new byte[6];
-            var channel = Channel.CreateUnbounded<ReadOnlyMemory<byte>>(
-                new UnboundedChannelOptions
-                {
-                    AllowSynchronousContinuations = false,
-                    SingleReader = true,
-                    SingleWriter = false
-                });
-
-            _channelReader = channel.Reader;
-            _channelWriter = channel.Writer;
+            _frameHeaderBuffer = new byte[8];
```

Comment:
Why 8? Thought we always read 7 bytes.

Reply:
Was planning on filling the 8th when validating the protocol but wasn't needed. Doesn't make all the difference though.